### DISCUSSION TITLE
REQUEST: Hypseus or Hypseus Singe instead of the old gcw0 port of Daphne

### DISCUSSION THREAD

**Discussion by mikepruett3** [eduardofilo/RG350_adam_image#80] [Ports]
I am not sure if its entirely possible, but I would love a more modern/updated version of Daphne on the RG350 devices. With the move to OpenDingux Beta, I feel like the handheld should be able to use these more modern emulators.

Hypseus - https://github.com/btolab/hypseus
Hypseus Singe - https://github.com/DirtBagXon/hypseus-singe

Thanks!

**Comment by eduardofilo** [comment]
I am sorry. My skills compiling C code are ridiculous. I only aspire to integrate the developments of others into the image.
I leave the issue open in case someone sees it and wants/can collaborate to compile those programs.

**Comment by mikepruett3** [comment]
Oh ok cool. I really need to learn how to cross-compile with the OpenDingux Beta toolchain.

**Comment by ghost** [comment]
Cross compiling is never a easy task. The point is not to follow a single sequence of commands and voila' everything is ported in a different platform. Also libraries shall be compatibles, etc...

I can give it a try to cross compile, but the chance of succeeding will be not so high, but I can try in my spare time...

**Comment by mikepruett3** [comment]
So I did try this already... actually created a Issue on DirtBagXon's repo page (https://github.com/DirtBagXon/hypseus-singe/issues/46#event-5566171562). He was Super Helpful!

I was able to get it to compile, but none of the SDL input keycode combinations seem to work with my hypinput.ini. I do not know how to translate the SDL INPUT environment variable to what is required in the hypinput.ini

**Comment by ghost** [comment]
Super Mike!

it would be nice if you could share what you did on a git to be replicated by others. I was about to give it a try myself, but not before my Xmas break for lack of time.

For the SDL INPUT honestly I do not know, I guess (but only guess) that some mapping shall be done, one suggestion could be to check ho the RG350 port of Daphne manages the input keycode. If I am not wrong Singe shall be a fork of Daphne, am I right?

**Comment by DirtBagXon** [comment]
> I was able to get it to compile, but none of the SDL input keycode combinations seem to work with my hypinput.ini. I do not know how to translate the SDL INPUT environment variable to what is required in the hypinput.ini

Not certain why the SDL2 keycodes are not recognised, SDL should automatically handle that for you.

Are you trying to use the mapped MACRO's (_i.e. SDLK_UP_) in _hypinput.ini_ ?

If so, you could try using the numeric (decimal) equivalents from:  https://github.com/DirtBagXon/hypseus-singe/blob/master/doc/keylist.txt

Look here for old style (pre MACRO) config: https://github.com/DirtBagXon/hypseus-singe/blob/66736810e025baaf7391da14fe763398c6b049a5/doc/hypinput.ini

The MACRO's should now just return the decimal values from the SDL library defined MACROS, so if that works maybe your SDL2 implementation is somehow limited....

If decimal values don't work, then you would need to look further into your SDL2 implementation:

```
case SDL_KEYUP:  
case SDL_KEYDOWN:
```

**Edit:** If I misunderstood your meaning of 'SDL input' and you meant the `joystick` input codes, then `jstest` is your route to figuring the correct values for the third field in config.

As @fantinomaurizio commented, if you have a previous RG350 port of Daphne, then the joystick (_SDL_) codes in that config  **should** be unchanged (_the 3rd field_), so firstly try copying them over to the third field in _hypinput.ini_ to the respective config entry.

**Reply by ghost** [reply]
Hello any update on Singe port on the RG350 @mikepruett3 ? I would love to test it, I like Daphne and I have no familiarity with Singe, but since Daphne seems limited I would love to give it a try to Singe and play it a little bit with new games.

**Comment by eduardofilo** [comment]
I transfer this conversation to the discussions section, since it is not an problem of the image itself.
